src/plugins/intel_cpu/tests/functional/shared_tests_instances/snippets/add.cpp

```diff
@@ -26,41 +32,77 @@ INSTANTIATE_TEST_SUITE_P(smoke_Snippets_Eltwise, Add,
                              ::testing::Values(1), // Subgraph is created, since the inputs are followed by converts
                              ::testing::Values(CommonTestUtils::DEVICE_CPU)),
                          Add::getTestCaseName);
+
+// DS
+std::vector<InputShape> inShapesDynamic1{
+        {
+        {{ov::Dimension::dynamic(), ov::Dimension::dynamic(), ov::Dimension::dynamic(), ov::Dimension::dynamic()},
+        {{1, 3, 1, 10}, {1, 3, 10, 10}, {1, 3, 1, 10}}},
+        }
+};
+std::vector<InputShape> inShapesDynamic2{
+        {
+        {{ov::Dimension::dynamic(), ov::Dimension::dynamic(), ov::Dimension::dynamic(), ov::Dimension::dynamic()},
+        {{1, 3, 10, 1}, {1, 3, 1, 1}, {1, 3, 10, 1}}},
+        }
+};
+INSTANTIATE_TEST_SUITE_P(smoke_Snippets_Eltwise_Add, Add,
+                         ::testing::Combine(
+                                 ::testing::ValuesIn(inShapesDynamic1),
+                                 ::testing::ValuesIn(inShapesDynamic2),
+                                 ::testing::Values(ov::element::f32),
+                                 ::testing::Values(1),
+                                 ::testing::Values(1), // Subgraph is created, since the inputs are followed by converts
+                                 ::testing::Values(CommonTestUtils::DEVICE_CPU)),
+                         Add::getTestCaseName);
+
+// ===================================AddPair=========================================================//
 // test cross-tile (vector vs scalar) optimizations in the absence of vector tile
-std::vector<std::vector<ov::Shape>> inShapesStatic{
-        {{1, 128, 1, 1}, {1, 128, 1, 1}},
-        {{1, 128, 1, 9}, {1, 128, 1, 9}},
-        {{1, 128, 1, 16}, {1, 128, 1, 16}},
-        {{1, 128, 1, 17}, {1, 128, 1, 17}},
-        {{1, 128, 1, 29}, {1, 128, 1, 29}},
-        {{1, 128, 1, 33}, {1, 128, 1, 33}},
-        {{1, 128, 9, 30}, {1, 128, 1, 30}},
-        {{1, 128, 9, 1}, {1, 128, 1, 30}},
-        {{1, 128, 9, 16}, {1, 128, 9, 1}},
+std::vector<std::vector<InputShape>> inShapesAddPair {
+        {{{}, {{1, 128, 1, 1}}}, {{}, {{1, 128, 1, 1}}}},
+        {{{}, {{1, 128, 1, 9}}}, {{}, {{1, 128, 1, 9}}}},
+        {{{}, {{1, 128, 1, 16}}}, {{}, {{1, 128, 1, 16}}}},
+        {{{}, {{1, 128, 1, 17}}}, {{}, {{1, 128, 1, 17}}}},
+        {{{}, {{1, 128, 1, 29}}}, {{}, {{1, 128, 1, 29}}}},
+        {{{}, {{1, 128, 1, 33}}}, {{}, {{1, 128, 1, 33}}}},
+        {{{}, {{1, 128, 9, 30}}}, {{}, {{1, 128, 1, 30}}}},
+        {{{}, {{1, 128, 9, 1}}}, {{}, {{1, 128, 1, 30}}}},
+        {{{}, {{1, 128, 9, 16}}}, {{}, {{1, 128, 9, 1}}}},
+        // DS
+        {{{1, -1, {1, 10}, {1, 33}}, {{1, 128, 1, 1}, {1, 128, 1, 9}, {1, 128, 1, 17}, {1, 128, 1, 29}, {1, 128, 9, 1}, {1, 128, 1, 1}}},
```

Comment:
Could you please add tests with sequence of the same shapes? To validate how Snippets work if we provide the same input shapes several times in a row

Reply:
added, thanks Alexandra!